File: gitdot-api-ts/src/resource/oauth.ts
import { z } from "zod";

export const DeviceCodeResource = z.object({
  device_code: z.string(),
  user_code: z.string(),
  verification_uri: z.string(),
  expires_in: z.number(),
  interval: z.number(),
});
export type DeviceCodeResource = z.infer<typeof DeviceCodeResource>;

export const TokenResource = z.object({
  access_token: z.string(),
  user_name: z.string(),
  user_email: z.string(),
});
export type TokenResource = z.infer<typeof TokenResource>;

File: gitdot-api/src/resource/oauth.rs
use serde::{Deserialize, Serialize};

#[derive(ApiResource, Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub struct DeviceCodeResource {
    pub device_code: String,
    pub user_code: String,
    pub verification_uri: String,
    pub expires_in: u64,
    pub interval: u64,
}

#[derive(ApiResource, Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub struct TokenResource {
    pub access_token: String,
    pub user_name: String,
    pub user_email: String,
}
